src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -430,6 +433,9 @@ private[immutable] object MapNode {
 
   def empty[K, V]: BitmapIndexedMapNode[K, V] = EmptyMapNode.asInstanceOf[BitmapIndexedMapNode[K, V]]
 
+  def single[K, V](key: K, value: V, keyHash: Int, originalHash: Int): BitmapIndexedMapNode[K, V] =
+    new BitmapIndexedMapNode[K, V](Node.bitposFrom(Node.maskFrom(keyHash, 0)), 0, Array(key, value), Array(originalHash), 1, keyHash)
```

Comment:
No need for a call to `Node.maskFrom(keyHash, 0))`, we can simplify to `Node.bitposFrom(keyHash)`